### PR TITLE
[kjobctl] Fix --mem-per-gpu flag

### PR DESCRIPTION
**Pull request by IrvingMg**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Fix issue that passes the wrong value to GPUs parser when using `--mem-per-gpu`.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cmd/experimental/kjobctl/pkg/builder/slurm_builder.go [hunk 1]
```diff
@@ -121,6 +121,10 @@ func (b *slurmBuilder) validateGeneral(ctx context.Context) error {
 		return noScriptSpecifiedErr
 	}
 
+	if b.memPerGPU != nil && b.gpusPerTask == nil {
+		return noGpusPerTaskSpecifiedErr
+	}
```
[the remark this answers is not in the document]
Comment (IrvingMg) [reply cmd/experimental/kjobctl/pkg/builder/slurm_builder.go:126]: Done.

**Comment by mbobrovskyi** [comment]
Could you please resolve conflicts?

**Comment by mbobrovskyi** [comment]
/lgtm

Thank!

**Pull request merged**